src/main/java/org/apache/sysml/utils/NativeHelper.java

```diff
@@ -198,7 +198,19 @@ private static boolean loadBLAS(String blas, String optionalMsg) {
 		}
 	}
 
-	private static boolean loadLibraryHelper(String path)  {
+	/**
+	 * Loads the specified library.
+	 * 
+	 * @param libName can be systemml_mkl or systemml_openblas
+	 * @return true if the library is loaded successfully
+	 */
+	private static boolean loadLibraryHelper(String libName)  {
+		// Default path assume x86_64 Linux
+		String path = "lib" + libName + "-Linux-x86_64.so";
```

Comment:
Since we might have to support ppc64le, its best to examine either `SystemUtils.OS_ARCH` or the `"os.arch"` property in Java. Please take a look at [how](https://github.com/os72/protoc-jar/blob/master/src/main/java/com/github/os72/protocjar/PlatformDetector.java#L182) protoc-jar does it. It should provide a good staring point.

Reply:
Sure, we can do that. In fact, we had that logic in this file (please check the history) :)

Reply:
I did, but what I am suggesting is to look for a file with the name:
`"lib" + "libName" + "<os>-<arch>" + ".<file_extension>"` or something close.